docs/how-to/create-device.md

```diff
@@ -1,32 +1,66 @@
+For example, only use `XYStage` for two perpendicular motors (e.g. X and Y axes on a sample table):
+- Do not use `XYStage` for unrelated motors or for motors that move in the same axis (e.g. coarse and fine adjustment).
+- Only device that is a group of motors with a physical relationship, should be defined in `motor`.
+- If your class define an `XYStage` but you need extra signals or behaviour, extend the `XYStage` class outside the `motor` module.
 
-If a device class is incompatible due to differences in PV address only, first request that an alias is added to the EPICS support module for the IOC - or request support in making that change. Only if it is not possible to add an alias, for example the support module is proprietary, add configurability to the dodal device class, taking care not to break existing devices- e.g. make new fields have a default that matches the existing pattern, and ensure that `dodal connect` is still able to connect to the device for the existing instances.
 
-If the device class is sufficiently different (or you cannot find a similar device), create a device that connects to the required signals and can be tested for your desired behaviour. During the review process, attempts to bring it closer in line with existing devices may allow to deduplicate some parts, through inheritance or composition of the device from existing components.
+Device classes that take a list of same device type may use ophyd-async's `DeviceVector`.  
+- Avoid dynamic attribute assignment (e.g. dicts of motors) as it hinders type checking and plan writing.  
+- Use static attributes and type hints for better IDE support and maintainability.
 
+If a compatible device class exists:
+- Use it and add it to the [beamline](./create-beamline.rst) to avoid re-implementation and share improvements.  
+- If a device class only differs by PV address, request an alias in the EPICS IOC support module.  
+- If that's not possible (e.g. proprietary support), add configurability to the dodal device class, ensuring defaults match existing patterns and that `dodal connect` still works for current devices.
+
+**Only if no suitable class exists**, create a new device that connects to the required signals. During review, refactor to align with existing devices if needed, using inheritance or composition to deduplicate code.  
 
 Writing a device class
 ======================
 
-The aim should be to get a new device ready for testing it on the beamline as soon as possible, to ensure fast iteration: write a device against your assumptions of how it should work, write tests against those assumptions then test your assumptions on the beamline. Write issues from beamline testing, to resolve offline to reserve as much time for testing that requires the beamline as possible.
+To develop a new device, aim for rapid prototyping and early beamline validation:
```

Comment:
Should: I think explicitly say that people should aim to get an early version into `main` as quickly as possible, test it on the beamline and iterate from there, always getting their changes into `main` early rather than letting PRs hang for months.